scripts/irods/test/test_ipasswd.py

```diff
@@ -139,7 +139,7 @@ def test_ipasswd_rodsuser_should_fail(self):
         #
         expected_lines = [
                             'CAT_INVALID_AUTHENTICATION',
-                            '826000 CAT_INVALID_AUTHENTICATION'
+                            '826000'
```

Comment:
Why does this need to be changed?
Did the output change?

Reply:
Yes this is a product of moving from `printError` to `printErrorStack` or the new `print_error_stack_to_file` functions.

The old output would look like this:
```
Error occurred while authenticating user [rods] [CAT_INVALID_AUTHENTICATION: failed to perform request

] [ec=-826000] failed with error -826000 CAT_INVALID_AUTHENTICATION
```

The new output looks like this:
```
Level 0: Error occurred while authenticating user [rods] [CAT_INVALID_AUTHENTICATION: failed to perform request

] [ec=-826000]
```

We lose the extra "failed with error -826000 CAT_INVALID_AUTHENTICATION" which In my opinion was redundant anyway.

Reply:
Is it just the message that has changed?

Reply:
Yeah. The error message changed so the tests are invalid.

Reply:
Okay, thinking out loud ...

It's an unstructured error message for the most part. Therefore, users should not rely on the message to automate anything. They need to react to the error code returned from the program. The error code being the code returned from `main()`.

With that said, I think this is fine.

@trel @alanking Thoughts?

Reply:
Does this affect anything outside of `ipasswd`? If not, then I think it's okay. If so, then it may be worth reconsidering/adjusting something to make the message the same.

I feel that `ipasswd` being solely affected is well-scoped enough to be manageable.

Reply:
It affects almost all icommands if a user is not logged in. 

Currently if a user is not logged in, they will receive the first message. That being said, `test_ipasswd` is the only test that looks for that message.

Reply:
it's still going to check for the full string, just alone, without the space in the literal string.

feels okay to me.   i'll be surprised to hear from anyone who has a script that is parsing/caring about return codes for logged-out attempts.

Reply:
Ah, that's right. This is mostly matters for failed login attempts.

Everything else using `printErrorStack` will remain unchanged.

@Nishant2022 Is that correct?

Reply:
That's right. Everything that uses `printErrorStack` will remain unchanged. In this PR I am only adding the ability to pass the client login errors on failed logins through the error stack, and as a result, we need to print them to `stderr` from the icommands to maintain the same functionality. The same amount of information is being passed through, but it is in a slightly different format.

Reply:
Excellent. Thanks.